userspace/libsinsp/sinsp.cpp

```diff
@@ -1284,11 +1268,52 @@ int32_t sinsp::next(OUT sinsp_evt **puevt)
 			evt->m_dump_flags = m_replay_scap_flags;
 			m_replay_scap_evt = NULL;
 		}
-		else 
+		else
```

Comment:
If i understood correctly we want to fetch an event from both `m_delayed_scap_evt` or `m_pending_state_evts` and give priority to `m_pending_state_evts`, right?
I would love a detailed comment here explaining the big picture :)

Reply:
Not really, the goal is to fetch the one with the "earliest timestamp" among the two. I'm gonna refactor to make it more clear.